api/classes/publisher.js

```diff
@@ -178,6 +178,10 @@ function updateProjectDetails() {
 function setRemixDataForPublishedProject() {
   var self = this;
 
+  if (self.project.readonly) {
```

Comment:
(2/2) and then not do this, but build the real meta data object, and assign the `readonly` flag as `true`.